README.md

```diff
@@ -1,2 +1,213 @@
+
+### Practical 5 - Build a Docker image
+
+**Expected -** Docker image should be available on GitLab Registry.
+
+- Set up [VueJs- admin panel](https://github.com/canopas/web-developer-roadmap-2023#practical-18) project if one doesn't already exist
+
+- Write GitLab CI script to run docker build
+  - Create a repository on GitLab if one doesn't already exist
+  - Write a docker file which includes,
+    - Install dependencies
+    - Build and run project
+  - Write a CI/CD script for Building a docker image and push it on the GitLab Registry.
+
+### Practical 6 - AWS EC2 Deployment
+
+**Expected -** Admin panel successfully run on `<ec2-ip-address>:<docker-container-port>`.
+
+- Perform following steps on AWS console
+  - Create EC2 instance with Security rules
+  - Install docker and enable docker swarm on the EC2 instance
+
+- Create a docker-compose file for the above docker image
+
+- Write GitLab CI script to deploy docker image on EC2 instance using docker-compose
+
+### Practical 7 - Configure Proxy and DNS Hosting
+
+**Expected -** Admin panel should run on DNS(i.e - admin.example.com).
+
+- Configure NGINX for [VueJs- admin panel](https://github.com/canopas/web-developer-roadmap-2023#practical-18).
+  - Create Self-signed SSL certificates for testing and configure them in NGINX
+  - Write a rule to allow http to https redirection
+  - Add reverse proxy for your project with DNS
+
+- Add NGINX configuration in Docker-compose which performs,
+  - Copy certificates to etc/nginx/certs on EC2
+  - Copy config files to etc/nginx/conf.d on EC2
+
+- Write CI/CD script for running nginx container on EC2 instance
+
+- Add DNS(i.e - admin.example.com) on Cloudflare and proxy it to NGINX
+
+## Static site deployment
+
+### Practical 8 - Deploy Nuxt static website on AWS S3 with Cloudfront
+
+**Expected -** The Website should run on the Cloudfront domain name.
+
+- Set up a nuxt-project if one doesn't already exist
+- Create SPA using a Static-site generator in Nuxt
+
+- Create S3 Bucket and Cloudfront distribution on the AWS console
+- Point Bucket link to Cloudfront origin
+
+- Write GitHub CI/CD to Generate a static website and deploy it on AWS S3
+
+## API deployment on AWS Lambda and API gateway
+
+### Practical 9 - Deploy golang api on aws lambda with route53 hosting using CloudFormation stack
+
+**Expected -** API should be accessible from Route53 DNS
+- Set up Golang API project using Gin if one doesn't already exist. ([Practical 22](https://github.com/canopas/web-developer-roadmap-2023#practical-22))
+- Use AWS SDK and Prepare code for Golang to deploy it to Lambda
+
+- Write GitHub CI/CD script for deploying cloudformation stack which includes,
+  - Define IAM's role and policies to provide required access
+  - Create AWS Lambda function for Golang
+  - Create an API gateway that can access the above lambda function
+  - Create Custom domain and API mapping
+  - Configure Route53 DNS for API gateway proxy
```

Comment:
As an additional step, it can include adding Eventrule as an alternative of CRON.

Reply:
That is an advanced thing. The purpose of this should be to provide basic idea about services